policy/modules/admin/fapolicyd.te

```diff
@@ -0,0 +1,160 @@
+policy_module(fapolicyd)
+
+########################################
+#
+# Declarations
+#
+
+attribute_role fapolicyc_roles;
+
+# for the fapolicyd daemon - long running process
+type fapolicyd_t;
+type fapolicyd_exec_t;
+init_daemon_domain(fapolicyd_t, fapolicyd_exec_t)
+
+# for the fapolicyd-cli process - interact with daemon
+type fapolicyc_t;
+type fapolicyc_exec_t;
+application_domain(fapolicyc_t, fapolicyc_exec_t)
+role fapolicyc_roles types fapolicyc_t;
+
+# for the fagenrules script - compile rules
+type fagenrules_t;
+type fagenrules_exec_t;
+application_domain(fagenrules_t, fagenrules_exec_t)
+init_script_domain(fagenrules_t, fagenrules_exec_t)
+
+type fapolicyd_config_t;
+files_config_file(fapolicyd_config_t)
+
+type fapolicyd_compiled_rules_t;
+files_type(fapolicyd_compiled_rules_t)
+
+type fapolicyd_log_t;
+logging_log_file(fapolicyd_log_t)
+
+type fapolicyd_runtime_t;
+files_runtime_file(fapolicyd_runtime_t)
+
+type fagenrules_tmp_t;
+files_tmp_file(fagenrules_tmp_t)
+
+type fapolicyd_unit_file_t;
+init_unit_file(fapolicyd_unit_file_t)
+
+type fapolicyd_var_lib_t;
+files_type(fapolicyd_var_lib_t)
+
+########################################
+#
+# fapolicyd (daemon) local policy
+#
+
+allow fapolicyd_t self:capability { audit_write chown dac_override setgid setuid sys_admin sys_nice sys_ptrace };
+allow fapolicyd_t self:process { setcap setsched };
+
+manage_fifo_files_pattern(fapolicyd_t, fapolicyd_runtime_t, fapolicyd_runtime_t)
+manage_files_pattern(fapolicyd_t, fapolicyd_runtime_t, fapolicyd_runtime_t)
+mmap_manage_files_pattern(fapolicyd_t,fapolicyd_var_lib_t,fapolicyd_var_lib_t)
+
+mmap_read_files_pattern(fapolicyd_t,fapolicyd_config_t,fapolicyd_config_t)
+files_runtime_filetrans(fapolicyd_t, fapolicyd_runtime_t, { file fifo_file })
+
+read_files_pattern(fapolicyd_t, fapolicyd_config_t, fapolicyd_compiled_rules_t)
+
+fapolicyd_read_config_files(fapolicyd_t)
+
+domain_read_all_entry_files(fapolicyd_t)
+domain_read_all_domains_state(fapolicyd_t)
+
+files_read_all_files(fapolicyd_t)
+files_read_all_symlinks(fapolicyd_t)
+files_map_usr_files(fapolicyd_t)
+files_watch_all_mountpoints(fapolicyd_t)
+
+fs_getattr_xattr_fs(fapolicyd_t)
+
+kernel_getattr_proc(fapolicyd_t)
+kernel_read_kernel_sysctls(fapolicyd_t)
+kernel_read_crypto_sysctls(fapolicyd_t)
+
+logging_log_filetrans(fapolicyd_t, fapolicyd_log_t, file)
```

Comment:
Seems that `fapolicyd_log_t` is dead, as this is the only access from inside this module.

Reply:
Sorry, I'm not sure what you are suggesting here.  There is a log file created.  Maybe the admin interface should grant read access to this type.  Maybe logrotate needs access (maybe the fact it is a log file grants logrotate necessary permissions?).

Reply:
Or maybe you are suggesting that there is no write access granted, so something is missing.  Let me check into that.

Reply:
Yup, it looks like it only generates the log file when 'do_stat_report' is enabled in the config file AND is exiting.  But, yes, I will all the rules to write to the log file. I see the denials for the missing rules.